### DISCUSSION TITLE
How to reorder things?

### DISCUSSION THREAD

**Discussion by ams1** [cloudscape-design/components#683] [Q&A] [answered]
Hi,

First of all I want to say a BIG thank you for open-sourcing this: THANK YOU! 😄 

I had a look over the components, but couldn't find anything that allows reordering (ex. via drag and drop or...).
Come to think of it, I also don't remember seeing reordering anywhere in the AWS console either (except the global nav favorites which allows reorder).

So the question: is there any plan for a "reordering pattern" in the future / what would be your guidelines regarding reordering (ex. I want to allow the user to reorder the columns of a table)

![image](https://user-images.githubusercontent.com/10154357/215711933-fb02ff03-575e-42f3-9777-38d204573ed4.png)

**Comment by YueyingLu** [comment]
Hey @ams1 ,

Thank you for reaching out. This feature would be a reasonable addition to the Cloudscape Design System. However, we don’t have a plan for this change in the short-term future. 

Thank you,
Yueying

**Comment by jperals** [comment, marked as answer]
Hey @ams1,

This feature is now available, starting from [version 3.0.363](https://github.com/cloudscape-design/components/releases/tag/3.0.263) which was released last week. To use it, provide the `columnDispay` prop of the [Table component API](https://cloudscape.design/components/table/?tabId=api) in combination with the `contentDisplayPrefence` prop of the [Collection preferences coomponent API](https://cloudscape.design/components/collection-preferences/?tabId=api). You can also find code examples by checking out the source code of any our demos that contain tables, such as [this one](https://cloudscape.design/examples/react/table.html).
